src/uu/dd/src/dd.rs

```diff
@@ -258,16 +258,37 @@ impl Source {
                         return Ok(len);
                     }
                 }
-                match io::copy(&mut f.take(n), &mut io::sink()) {
-                    Ok(m) if m < n => {
+                // Try seek first; fall back to read if not seekable
+                match n.try_into().ok().map(|n| f.seek(SeekFrom::Current(n))) {
+                    Some(Ok(pos)) => {
+                        if pos > f.metadata().map(|m| m.len()).unwrap_or(u64::MAX) {
+                            show_error!(
+                                "{}",
+                                translate!("dd-error-cannot-skip-offset", "file" => "standard input")
+                            );
+                        }
+                        Ok(pos)
```

Comment:
Returns wrong value. Should return `Ok(n)` (bytes skipped) not `Ok(pos)` (absolute file position). no ?